src/main/java/org/colonelkai/publictransit/line/Line.java

```diff
@@ -82,16 +105,31 @@ public String getIdentifier() {
         return this.identifier;
     }
 
-    public String getName() {
-        return this.name;
+    public AText getName() {
+        return AText.ofPlain(this.name); //needs fixing
     }
 
     public List<Node> getNodes() {
         return this.getNodes(false);
     }
 
     public List<Node> getNodes(boolean reverse) {
-        List<Node> nodes = new ArrayList<>(this.nodes);
+        return this.applyReverseLogic(this.nodes, reverse);
+    }
+
+    public List<Node> getNodesUnmodified() {
+        return Collections.unmodifiableList(this.nodes);
+    }
+
+    public void addNode(Node node) {
+        Collection<Node> nodes = new ArrayList<>(this.nodes);
+        nodes.add(node);
+        this.validate(nodes);
+        this.nodes.add(node);
+    }
+
+    private List<Node> applyReverseLogic(List<Node> nodesList, boolean reverse) {
```

Comment:
From what I understand, this makes it so that we get the reversed version only if the parameter OR the `oneWayReversed` is true, not if neither or both. 

I have a feeling that 'applyReverseLogic' seriously overcomplicates the matter and I'm still on the side that we should internally keep the nodes always at the same orientation. 

But, for the sake of consistency, and user intuitiveness, we HAVE to keep the front-end stack the same at all times, meaning the indexes cannot change and thus cannot be reversed. 

My problem here is that we would need to reverse the list in only 1 case; if the player is traveling NEGATIVELY through a line, or planning to do so.

The whole thing with oneWayReversed is that the player can ONLY travel NEGATIVELY. We don't need to have such a system for something so simple, I'd rather remove this and just literally never reverse the nodes and just loop negatively.

Reply:
Personally would disagree. 

The check there is if the reversed parameter is true and the reversed one way is true, or the reverse parameter is false and the reversed one way is false then dont bother reversing the list as it would be reversed and then reversed again, so this is more a efficiency check. 

When it comes to anything user facing you would use the ``getNodesUnmodified`` as that maintains the original order. While anything logical we would use this.

My problem with not reversing the list is if one of the two edge cases you mentioned happens, there needs to be added code outside this class (class dependency issue) to correct for that, which typically leads to extra code, duplicated code and therefore more bugs

Reply:
We don't need to reverse it if `oneWayReversed` is true. It's NOT a modifier to which side a player goes, its a condition on which side a player CAN go.

Reply:
How about we only have a `getNodes()` with a `reversed` parameter? This would cover both edge cases I said and like I stated, we would NOT need to reverse it if `oneWayReversed` is true. 

And actually, we don't need to deal with `oneWayReversed` outside of this class AT ALL. We can just create a `getReachableNodes()` function that we pass the current node onto, and it returns us all nodes that we can reach from that node. 

This way, we can enforce our checks that are dependent on `oneWay` and `oneWayReversed` (which is only being able to go one way, and then only being able to go one way in REVERSE) entirely within the class.